### PR TITLE
setting imagePullSecrets for workers

### PR DESCRIPTION
**Pull request by chokosabe**
Setting the imagepullsecrets. Not sure this is the best location to do this in.

### CODE REVIEW

**Review by mwylde — CHANGES_REQUESTED** [review]
This is a great start but need a little more work

**Comment by chokosabe** [comment]
Added the changes requested. There does seem to be an intermittent build error with mqtt.

**Comment by chokosabe** [comment]
Updates pushed up. Resolved all the changes.

File: k8s/arroyo/templates/configmap.yaml [hunk 1]
```diff
@@ -83,4 +87,4 @@ data:
           - name: arroyo-user-config
             mountPath: /config
           {{- end }}
-        command: "/app/arroyo-bin --config-dir /config worker"
\ No newline at end of file
+        command: ["/app/arroyo-bin --config-dir /config worker"]
```
Comment (mwylde) [inline k8s/arroyo/templates/configmap.yaml:90]: This still isn't quite right. With this, trying to run this in helm gives the error 

```
Configuration is invalid!
  • invalid type: found sequence, expected a string for key "default.kubernetes-scheduler.worker.command" in /root/.config/arroyo/config.yaml YAML file
```
It should be restored to what it currently is in master.

File: k8s/arroyo/templates/configmap.yaml [hunk 2]
```diff
@@ -59,6 +59,10 @@ data:
           {{- end }}    
         image: "{{ .Values.worker.image.repository }}:{{ .Values.worker.image.tag }}"
         image-pull-policy: "{{ .Values.worker.image.pullPolicy }}"
+        image-pull-secrets:
+        {{- with .Values.imagePullSecrets }}
```
Comment (mwylde) [inline k8s/arroyo/templates/configmap.yaml:63]: `image_pull_secrets` is defined as a string, but here it's passed in an object which fails:
```
Configuration is invalid!
  • invalid type: found unit, expected a string for key "default.kubernetes-scheduler.worker.image-pull-secrets" in /root/.config/arroyo/config.yaml YAML file
```

File: crates/arroyo-rpc/default.toml [hunk 3]
```diff
@@ -64,6 +64,7 @@ resource-mode = "per-slot"
 name-prefix = "arroyo"
 image = "ghcr.io/arroyosystems/arroyo:latest"
 image-pull-policy = "IfNotPresent"
+image-pull-secrets = "IfNotPresent"
```
Comment (mwylde) [inline crates/arroyo-rpc/default.toml:67]: Is this a sensible default for this?

**Review by mwylde — CHANGES_REQUESTED** [review]
This still isn't quite working. It might be useful to try testing it locally with docker desktop. With a local k8s environment, you can test it by creating a values.yaml file like this:

```
artifactUrl: "/tmp/arroyo-test"
checkpointUrl: "/tmp/arroyo-test"

volumes:
  - name: checkpoints
    hostPath:
      path: /tmp/arroyo-test
      type: DirectoryOrCreate

volumeMounts:
  - name: checkpoints
    mountPath: /tmp/arroyo-test
```

Then building an image from your branch and deploying it locally:

```
$ docker build . -f docker/Dockerfile -t ghcr.io/arroyosystems/arroyo:tip --target arroyo
$ helm install arroyo  k8s/arroyo -f values.yaml 
```

**Review by mwylde — APPROVED** [review]
Thanks for the contribution! I've gone ahead and rebased on master and fixed a couple of issues so that we can get this into the next release.

**Pull request merged**